plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -563,12 +563,12 @@ private ListenableFuture<Void> loadPartition(HivePartitionMetadata partition)
 
     private List<TrinoFileStatus> listBucketFiles(TrinoFileSystem fs, Location location, String partitionName)
     {
-        if (!ignoreAbsentPartitions) {
-            checkPartitionLocationExists(fs, location);
-        }
-
         try {
-            return ImmutableList.copyOf(new HiveFileIterator(table, location, fs, directoryLister, hdfsNamenodeStats, FAIL));
+            HiveFileIterator fileIterator = new HiveFileIterator(table, location, fs, directoryLister, hdfsNamenodeStats, FAIL);
```

Comment:
Will iterator creation report error if location does not exists. If so will this error be as nice as one generated by `checkPartitionLocationExists`?

Reply:
> Will iterator creation report error if location does not exists?

It will not, that's why the check is there.